device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -1113,6 +1124,22 @@ void setOption_SetAmqpOpenDeviceSessionsTimeout(Object value)
         }
     }
 
+    void setOption_SetMaxMessagesSentPerThread(Object value)
+    {
+        if (value != null)
```

Comment:
What if `value` is null?

Reply:
I'll make it throw IllegalArgumentException, sure